src/orderbook/state.ts

```diff
@@ -28,7 +83,139 @@ export class AccountState {
       )
     }
     this.batchId = batchId
-    // TODO(nlordell): Actually apply events to state.
+
+    for (const ev of events) {
+      const data = ev.returnValues
+      switch (ev.event) {
+        case 'OrderCancellation':
+          this.order(data.owner, parseInt(data.id)).validUntil = (await this.batches.forEvent(ev)) - 1
```

Comment:
Would be nice to pull this into a small method too to keep this switch statement free of logic.